### ISSUE TITLE
Rename all example files from .sil to .vpr

### ISSUE THREAD

**Issue by fpoli** [fpoli/viper-silver#114]
All the example files with .sil extension should be renamed to .vpr since the former is being deprecated. Changes in the code should also be considered if necessary. This task must include Silver, Silicon and Viper repositories.

Care should be taken not to lose the commit history of the files, which seems not to be automatically copied by “hg rename”.

Collisions should be avoided, as there are cases where two files have the same base name but differ in the extension \(sil and vpr\).

Imports, comments and readme files should also reflect such changes.

[Issue created by fabiopakk: 2019-08-05]
[Last updated on bitbucket: 2019-08-06]

[Comment created by fabiopakk: 2019-10-24]
Issue #149 was marked as a duplicate of this issue.

**Comment by fpoli** [comment]
> **@caterinaurban** commented on 2016-03-07 17:06

**Comment by fpoli** [comment]
> **@arshavir** commented on 2016-04-07 15:42

When the number of args in the function call exceeds the number of args in the definition, the fail looks like this:

```
function foo(x: Int): Bool
  ensures true


method bar()
{
   assert foo(1, 2)
}
```

```
java.lang.IndexOutOfBoundsException: 1
	at scala.collection.LinearSeqOptimized$class.apply(LinearSeqOptimized.scala:65)
	at scala.collection.immutable.List.apply(List.scala:84)
	at viper.silver.parser.PFunctApp$$anonfun$signatures$1.apply(ParseAst.scala:408)
	at viper.silver.parser.PFunctApp$$anonfun$signatures$1.apply(ParseAst.scala:408)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
...
```

There's a the same issue with domain functions.

**Comment by fpoli** [comment]
> **@arshavir** commented on 2016-04-07 15:46

This is not an issue for predicates. For them a nice message ```predicate arity doesn't match``` is generated.

**Comment by fpoli** [comment]
> **@arshavir** commented on 2016-04-07 15:49

**Comment by fpoli** [comment]
> **@meilers** commented on 2016-07-29 12:52

**Comment by fpoli** [comment]
> **@meilers** commented on 2016-07-29 12:52

Resolved in pull request #54.
